[Package]: libfixposix

### Why do you want to add this package to Termux?

The ultimate goal is to package the Nyxt web browser (based on WebKitGTK renderer).

`libfixposix` is a dependency of many widely used Common Lisp libraries.

### Home page URL

_No response_

### Source code URL

https://github.com/sionescu/libfixposix/

### Packaging policy acknowledgement

- [X] The requested package complies with [Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy).

### Additional information

Here's my attempt to package it. Notice that I'm building it from master since there's [a commit after the latest release that is extremely important to success](https://github.com/sionescu/libfixposix/commit/4be4a7a729f2f56e162d3d42b8d18bb811fd9438).

```shell
TERMUX_PKG_HOMEPAGE=https://github.com/sionescu/libfixposix/
TERMUX_PKG_DESCRIPTION="Thin wrapper over POSIX syscalls"
TERMUX_PKG_LICENSE="BSL-1.0"
TERMUX_PKG_MAINTAINER="@termux"
TERMUX_PKG_VERSION=0.5.1
TERMUX_PKG_GIT_BRANCH="master"
TERMUX_PKG_BUILD_DEPENDS="pkg-config, libtool, check"
TERMUX_PKG_SRCURL=git+https://github.com/sionescu/libfixposix

termux_step_pre_configure() {
	autoreconf -fi
}

```

When I try to build the package defined above I get stuck in building `termux-am`. 

Any idea @Grimler91? Thank you.

See the log below:

```log
~/repos/termux-packages $ ./scripts/run-docker.sh ./build-package.sh termux-am
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
termux - building termux-am for arch aarch64...
]0;termux-am...
<-------------> 0% INITIALIZING [16ms]
<-------------> 0% INITIALIZING [116ms]> Evaluating settings<-------------> 0% INITIALIZING [219ms]<-------------> 0% INITIALIZING [316ms]<-------------> 0% INITIALIZING [417ms]<-------------> 0% INITIALIZING [516ms]<-------------> 0% INITIALIZING [616ms]<-------------> 0% INITIALIZING [716ms]<-------------> 0% INITIALIZING [816ms]<-------------> 0% INITIALIZING [918ms]<-------------> 0% INITIALIZING [1s]<-------------> 0% CONFIGURING [1s]> Loading projects> root project<-------------> 0% CONFIGURING [2s]> root project > Resolve dependencies of :classpath<-------------> 0% CONFIGURING [3s]> root project > Resolve files of :classpath> root project<======-------> 50% CONFIGURING [3s]> :app<======-------> 50% CONFIGURING [4s]<======-------> 50% CONFIGURING [5s]<======-------> 50% CONFIGURING [6s]<=============> 100% CONFIGURING [6s]> IDLE
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> Could not create task ':app:assembleRelease'.
   > Could not create task ':app:lintVitalRelease'.
      > SDK location not found. Define location with an ANDROID_SDK_ROOT environment variable or by setting the sdk.dir path in your project's local properties file at '/home/builder/.termux-build/termux-am/src/local.properties'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 8s


<-------------> 0% WAITING> IDLE
```

--- a/packages/libfixposix/build.sh
+++ b/packages/libfixposix/build.sh
@@ -1,0 +1,15 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/sionescu/libfixposix/
+TERMUX_PKG_DESCRIPTION="Thin wrapper over POSIX syscalls"
+TERMUX_PKG_LICENSE="BSL-1.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=0.5.1
+TERMUX_PKG_SRCURL=https://github.com/sionescu/libfixposix/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=5d9d3d321d4c7302040389c43f966a70d180abb58d1d7df370f39e0d402d50d4
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+ac_cv_have_decl_TIOCSCTTY=yes
+ac_cv_prog_PKGCONFIG=yes
+"
+
+termux_step_pre_configure() {
+	autoreconf -fi
+}

--- a/packages/libfixposix/src-lib-stdlib.c.patch
+++ b/packages/libfixposix/src-lib-stdlib.c.patch
@@ -1,0 +1,30 @@
+--- a/src/lib/stdlib.c
++++ b/src/lib/stdlib.c
+@@ -142,6 +142,7 @@
+     return NULL;
+ }
+ 
++#ifndef __ANDROID__
+ // FIXME: add autoconf check that confstr(_CS_PATH) returns sane values
+ static char*
+ _lfp_default_path(void)
+@@ -151,6 +152,7 @@
+     confstr(_CS_PATH, default_path, default_path_size);
+     return default_path;
+ }
++#endif
+ 
+ DSO_PUBLIC char*
+ lfp_getpath(char *const envp[])
+@@ -160,7 +162,11 @@
+     if (envpath != NULL) {
+         return strdup(envpath);
+     } else {
++#ifndef __ANDROID__
+         return _lfp_default_path();
++#else
++        return "@TERMUX_PREFIX@/bin";
++#endif
+     }
+ }
+ 

--- a/packages/libfixposix/src-lib-strerror.c.patch
+++ b/packages/libfixposix/src-lib-strerror.c.patch
@@ -1,0 +1,12 @@
+--- a/src/lib/strerror.c
++++ b/src/lib/strerror.c
+@@ -24,6 +24,9 @@
+ 
+ #include <lfp/strerror.h>
+ 
++#ifdef __ANDROID__
++#undef __USE_GNU
++#endif
+ #include <string.h>
+ 
+ #if defined(HAVE___XPG_STRERROR_R)

